src/components/rooms-tab.js

```diff
@@ -35,6 +81,46 @@ function RoomsTab({ rooms }) {
     return <span>{t(`accessType_${access}`)}</span>;
   };
 
+  const handleNewRoomClick = () => {
+    const newRoom = createNewRoomState();
+    setState(prevState => ({ ...prevState, isNewRoomModalVisible: true, newRoom }));
+    if (newRoomFormRef.current) {
+      newRoomFormRef.current.setFieldsValue({ name: newRoom.name });
+    }
+  };
+
+  const handleNewRoomOk = async () => {
+    try {
+      const isValid = await isNewRoomFormValid();
+      if (!isValid) {
+        return;
+      }
+
+      setState(prevState => ({ ...prevState, isNewRoomBeingCreated: true }));
+      const newRoom = await roomApiClient.addRoom({ name: state.newRoom.name, access: state.newRoom.access });
+      setState(prevState => ({ ...prevState, isNewRoomBeingCreated: false, isNewRoomModalVisible: false }));
+
+      window.location = urls.getRoomDetailsUrl(newRoom._id);
+    } catch (error) {
+      errorHelper.handleApiError({ error, logger, t });
+      setState(prevState => ({ ...prevState, isNewRoomBeingCreated: false }));
```

Comment:
This one can go in a finally block so we never have a chance to freeze the UI.

Reply:
The UI won't freeze, this is set in the `try` block:

`setState(prevState => ({ ...prevState, isNewRoomBeingCreated: false, isNewRoomModalVisible: false }));`

Reply:
If the error helper throws, it will (I've had the situation happen with some mismatched parameters). But okay, it's overkill.